src/providers/interpretation_providers/word2vecInterpretationProvider.ts

```diff
@@ -29,29 +33,68 @@ export class Word2VecInterpretationProvider implements InterpretationProvider {
 		'cocktail',
 		'wine',
 	];
-	protected isAlcoholic(word: string): boolean {
-		return Word2VecInterpretationProvider.alcoholicWords.some(
-			(alcoholicWord) => this.similar(word, alcoholicWord)
-		);
+	protected static nonAlcoholicWords: ReadonlySet<string> = new Set([
+		'coffee',
+		'non-alcoholic',
+	])
+	protected isAlcoholic(text: string): boolean {
+		const words = text.toLowerCase().split(' ');
+		return (
+			!words.some((word) => Word2VecInterpretationProvider.nonAlcoholicWords.has(word))
+			&& words.some((word) =>
+				Word2VecInterpretationProvider.alcoholicWords.some((alcoholicWord) =>
+					this.similar(word, alcoholicWord)
+				)
+			)
+		)
 	}
 
-	public interpret(information: InterpretationParams): BusinessInsights {
-		// TODO: waiting on ExtractedImage for actual implementation
-		return {
-			hasDelivery: {
-				insight: true,
-				confidence: 0.7,
-				evidence: {
-					images: [],
-				},
-			},
+	protected inferImage(inferences: BusinessInferences, image: ExtractedImage): BusinessInferences {
+		const { assigned_labels, detected_objects } = image.extractions;
+		const keywords = new Map<string, Array<Extraction>>();
+		for (const label of assigned_labels) {
+			const { description: keyword } = label;
+			keywords.get(keyword)?.push(label) || keywords.set(keyword, [label]);
+		}
+		for (const obj of detected_objects) {
+			const { object_name: keyword } = obj;
+			keywords.get(keyword)?.push(obj) || keywords.set(keyword, [obj]);
+		}
+		console.debug(keywords);
+
+		// Infer alcoholism
+		const { servesAlcohol } = inferences;
+		const alcoholicKeywords = Array.from(keywords.keys()).filter(this.isAlcoholic.bind(this));
+		if (alcoholicKeywords.length > 0) {
+			const alcoholicExtractions = alcoholicKeywords.reduce(
+				(extractions, keyword) => extractions.concat(keywords.get(keyword)!),
+				[] as Array<Extraction>
+			);
+			servesAlcohol.insight = true;
+			servesAlcohol.confidence = alcoholicExtractions.reduce(
+				(confidence, x) => Math.max(confidence, x.confidence),
+				servesAlcohol.confidence
+			);
+			servesAlcohol.evidence.images.push({
+				source: image,
+				reason: `detected keywords: ${alcoholicKeywords.join(', ')}`,
+			});
```

Comment:
Merging this result with the other image inferences shouldn't be the responsibility of this function. It should be controlled by `interpret`.

Reply:
Defered to #30